Library/Homebrew/formula_installer.rb

```diff
@@ -1179,9 +1182,17 @@ def fetch_dependencies
     deps.each { |dep, _options| fetch_dependency(dep) }
   end
 
+  sig { returns(T.nilable(Formula)) }
+  def previously_fetched_formula
+    # We intentionally don't compare classes here.
+    self.class.fetched.find do |fetched_formula|
```

Comment:
Might be nicer to use a `Hash` instead of `Set` to make this sort of lookup a bit nicer.